src/Symfony/Bridge/Doctrine/Form/Type/EntityType.php

```diff
@@ -68,10 +68,12 @@ public function getBlockPrefix()
      * We consider two query builders with an equal SQL string and
      * equal parameters to be equal.
      *
+     * @param QueryBuilder $queryBuilder
```

Comment:
should we add `|Builder` from the ODM to ensure someone else won't readd the type later on? (same above)

Reply:
well, the thing is, the EntityType is specific to ORM. But DoctrineType is not.

Reply:
Then we should update the phpdoc in the `DoctrineType` too, and throw an exception in the EntityType to ensure the query builder class is from ORM. What about a test?

Reply:
I agree with @HeahDude - the test should reject anything that is not a `QueryBuilder` of course.
At least we need something that will prevent us from adding the type by mistake again :)

Reply:
Can you please release this fix and decide later on other stuff, because this bug is breaking my application. Thanks.

Reply:
> Then we should update the phpdoc in the DoctrineType too, and throw an exception in the EntityType to ensure the query builder class is from ORM. What about a test?

This is already done by the options resolver. I don't think we need more.

> At least we need something that will prevent us from adding the type by mistake again

I added a comment in the docblock

Reply:
Fair enough, but the normalizer handling the exception might be overridden by `EntityType` children. So I would still enforce the check inside the methods `getLoader()` and `getQueryBuilderPartsForCachingHash` of the `EntityType`.

Also we need another PR targeting 5.0 because the type hint is also present for the abstract method `getLoader` there, do you want me to open it @xabbuh or can you take care of that too? Thanks

Reply:
fair enough, checks added in both methods